server/portal/apps/workspace/api/views.py

```diff
@@ -333,6 +364,8 @@ def getPublicApps(self, user):
             # Retrieve all apps known to the portal in that directory
             tapis_apps = list(AppTrayEntry.objects.all().filter(available=True, category=category, appType='tapis')
                               .order_by(Coalesce('label', 'appId')).values('appId', 'appType', 'html', 'icon', 'label', 'version'))
+
+            tapis_apps = [x for x in tapis_apps if any(x['appId'] == y.id for y in apps_listing)]
```

Comment:
do we need to consider appVersion here **if a admin has defined an app with a appVersion in the model (AppTrayEntry) **

also, should we log an error or a warning if there a app entries that doesn't match up with the tapis apps?

Reply:
Good catch, will add accommodation for presence of appVersion in the id.

Hmm not sure how helpful those logs would be though